hub/core/meta/encode/shape.py

```diff
@@ -0,0 +1,147 @@
+from typing import Tuple
+from hub.core.storage.provider import StorageProvider
+import numpy as np
+
+
+SHAPE_ENCODING_DTYPE = np.uint64
+LAST_INDEX_INDEX = -1
+
+
+class ShapeEncoder:
+    def __init__(self, encoded_shapes: np.ndarray = None):
+        """Custom compressor that allows reading of shapes from a sample index without decompressing.
+        Requires that all shapes encoded have the same dimensionality (`len(shape)`).
```

Comment:
Is this requirement flexible?

Reply:
kind of

Reply:
we may allow shapes with a max dimensionality, and just auto-add 1s where necessary. like if we have a (28, 28, 3) image, but subsequent (28, 28, 1) images

Reply:
but we have discussed why this is a bad idea